tests/ui/projection/impl-projection-on-primitive.stderr

```diff
@@ -0,0 +1,11 @@
+error[E0390]: cannot define inherent `impl` for primitive types
+  --> $DIR/impl-projection-on-primitive.rs:13:6
+   |
+LL | impl <S as Identity>::Identity {
```

Comment:
Hm, I'm not convinced that this is behavior we *want*

Reply:
I kept the behavior when trying to implement something on a primitive (since when implementing on the projection, you actually implement on the type behind the projection itself, which is `()` here).

How should it behave otherwise?

Reply:
Sorry, I guess I left this on the wrong UI test. I'm more concerned about allowing us to (successfully) implement an inherent impl on a projection. Let me leave a new comment.